core/src/main/java/bio/terra/pearl/core/service/export/EnrolleeImportService.java

```diff
@@ -167,6 +149,65 @@ public Import importEnrollees(String portalShortcode, String studyShortcode, Stu
         return dataImport;
     }
 
+
+    // An account can have multiple enrollees. This
+    // record groups them together so our import process
+    // can create them all at once and link them together.
+    //
+    // Proxy accounts are determined by the presence of the
+    // proxy.email field in the import data.
+    @Getter
+    @Setter
+    public static class AccountImportData {
+        private String email;
+        private Map<String, String> enrolleeData = null;
+        private List<Map<String, String>> proxyData = new ArrayList<>();
+    }
+
+    List<AccountImportData> groupImportMapsByAccount(List<Map<String, String>> enrolleeMaps) {
+        List<AccountImportData> accountData = new ArrayList<>();
+
+        for (Map<String, String> enrolleeMap : enrolleeMaps) {
+            String email = enrolleeMap.get("account.username");
+            String proxyEmail = enrolleeMap.get("proxy.username");
+            if (email == null || email.isBlank()) {
+                email = proxyEmail;
+            }
+
+            AccountImportData existingAccount = findExistingAccount(accountData, email);
+            if (proxyEmail == null || proxyEmail.isBlank()) {
+                if (existingAccount == null) {
+                    AccountImportData newAccount = new AccountImportData();
+                    newAccount.setEmail(email);
+                    newAccount.setEnrolleeData(enrolleeMap);
+                    accountData.add(newAccount);
+                } else {
+                    existingAccount.setEnrolleeData(enrolleeMap);
+                }
+            } else {
+                if (existingAccount == null) {
+                    AccountImportData newAccount = new AccountImportData();
+                    newAccount.setEmail(email);
+                    newAccount.getProxyData().add(enrolleeMap);
+                    accountData.add(newAccount);
+                } else {
+                    existingAccount.getProxyData().add(enrolleeMap);
+                }
+            }
```

Comment:
Shorter and IMHO easier-to-read
```suggestion
            if (existingAccount == null) {
                AccountImportData newAccount = new AccountImportData();
                newAccount.setEmail(email);
                if (StringUtils.isBlank(proxyEmail) {
                  newAccount.setEnrolleeData(enrolleeMap);
                }  else {
                  newAccount.getProxyData().add(enrolleeMap);
                }
                accountData.add(newAccount);        
            } else {
                if (StringUtils.isBlank(proxyEmail) {
                    existingAccount.setEnrolleeData(enrolleeMap);
                } else {
                    existingAccount.getProxyData().add(enrolleeMap);
                }
            }
```